src/node_port.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling mongos Kubernetes resources for a single mongos pod."""
+
+import logging
+from functools import cached_property
+from ops.charm import CharmBase
+from lightkube.models.meta_v1 import ObjectMeta, OwnerReference
+from lightkube.core.client import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.core_v1 import Pod, Service
+from lightkube.models.core_v1 import ServicePort, ServiceSpec
+
+
+logger = logging.getLogger(__name__)
+
+# default logging from lightkube httpx requests is very noisy
+logging.getLogger("lightkube").disabled = True
+logging.getLogger("lightkube.core.client").disabled = True
+logging.getLogger("httpx").disabled = True
+logging.getLogger("httpcore").disabled = True
+
+
+class NodePortManager:
+    """Manager for handling mongos Kubernetes resources for a single mongos pod."""
+
+    def __init__(
+        self,
+        charm: CharmBase,
+    ):
+        self.charm = charm
+        self.pod_name = self.charm.unit.name.replace("/", "-")
+        self.app_name = self.charm.app.name
+        self.namespace = self.charm.model.name
+
+    @cached_property
+    def client(self) -> Client:
+        """The Lightkube client."""
+        return Client(  # pyright: ignore[reportArgumentType]
+            field_manager=self.pod_name,
+            namespace=self.namespace,
+        )
+
+    # --- GETTERS ---
+
+    def get_pod(self, pod_name: str = "") -> Pod:
+        """Gets the Pod via the K8s API."""
+        # Allows us to get pods from other peer units
+        pod_name = pod_name or self.pod_name
+
+        return self.client.get(
+            res=Pod,
+            name=self.pod_name,
+        )
+
+    def build_node_port_services(self, port: str) -> Service:
```

Comment:
In our setup, I don't see a need to expose this method (re: previous comment)
```suggestion
    def _build_node_port_services(self, port: str) -> Service:
```

Reply:
total noob question.

We call it in:
`src/charm.py:139:            service=self.node_port_manager.build_node_port_services(`

wouldn't we want it to be exposed for usage in `src/charm.py`?